tool/requote.js

```diff
@@ -51,6 +52,10 @@ function findSpanType(program, begin_index, end_index) {
     let spanType;
     if (begin_index > 1 && program[begin_index-2] === 'location:') {
         spanType = 'LOCATION';
+    } else if (program[begin_index-2] === '==' && program[begin_index-3].endsWith('Number')){
+        spanType = 'NUMBER';
+    } else if (program[begin_index-2] === '==' && program[begin_index-3].includes('phone_number')){
+        spanType = 'PHONE_NUMBER';
```

Comment:
No. You cannot rely on type annotations, nor you can rely on having a param: token nearby.
If you want to detect a number, use a regexp that detects digits.
Note that unquoted numbers in thingtalk **do not** the `"` character. That is, the syntax is
```
now => @org.schema.Restaurant(), param:reviewCount >= 7 => notify;
```
rather than
```
now => @org.schema.Restaurant(), param:reviewCount >= " 7 " => notify;
```
(this is already existing syntax - if you invent new syntax you need to change thingtalk too)

As for phone_number/email_address/url, you should do exactly what the existing code does that for username/hashtag (that is, append `^^` and the entity identifier after the string literal)